app/dashboard/dashboard-client.tsx

```diff
@@ -70,6 +87,35 @@ export default function DashboardClient({ docsList }: { docsList: any }) {
     router.push(`/document/${data.id}`);
   }
 
+  async function deleteDocument(id: string, fileUrl: string) {
+    try {
+      const res = await fetch('/api/deleteDocument', {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          id,
+          fileUrl,
+        }),
+      });
+
+      if (!res.ok) {
+        throw new Error('Failed to delete document');
+      }
+
+      const data = await res.json();
+      if (data.error) {
+        console.log(data.error);
+      } else {
+        console.log('Document deleted successfully');
+        updateDocsList(docsList.filter((doc) => doc.id !== id));
```

Comment:
Probably unnecessary to wrap this? Just call `setDocsList` directly?

Reply:
Yeah right. I have done all the requested changes